File: Day 11/Remove Duplicates from Sorted List.cpp
// Question Link ---> https://leetcode.com/problems/remove-duplicates-from-sorted-list/
// Day #11
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* walk = head;
        while (walk != NULL && walk->next != NULL) {
            if (walk->val == walk->next->val) walk->next = walk->next->next;
            else walk = walk->next;
        }
        return head;
    }
};

File: Day 11/Remove Duplicates from Sorted List.java
// Question Link ---> https://leetcode.com/problems/remove-duplicates-from-sorted-list/
// Day #11
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode walk = head;
        while (walk != null && walk.next != null) {
            if (walk.next.val == walk.val) {
                walk.next = walk.next.next;
            }
            else {
                walk = walk.next; 
            }
        }
        return head;
    }
}